src/rdb/rdb_raft.c

```diff
@@ -1855,9 +1965,11 @@ rdb_raft_append_apply_internal(struct rdb *db, msg_entry_t *mentry,
 			goto out;
 	}
 
+	ABT_mutex_lock(db->d_raft_mutex);
 	rdb_raft_save_state(db, &state);
 	rc = raft_recv_entry(db->d_raft, mentry, &mresponse);
 	rc = rdb_raft_check_state(db, &state, rc);
+	ABT_mutex_unlock(db->d_raft_mutex);
```

Comment:
Exactly. The "do not yield" comment should be removed.

Let's just remove the lock and unlock calls from rdb_raft_wait_applied then. It will look fine to me even without an additional comment.